src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java

```diff
@@ -56,7 +56,77 @@
  * <pre>
  * &lt;module name=&quot;IllegalCatch&quot;/&gt;
  * </pre>
+ * <pre>
+ * try {
+ *     // do stuff
+ * } catch (Exception e) { // Violation
+ *
+ * }
+ *
+ * try {
+ *     // do stuff
+ * } catch (ArithmeticException e) {
+ *
+ * } catch (Exception e) { // Violation, one of the exception is declared to be illegal
+ *                           and order of multi-catch blocks doesn't matter
+ *
+ * }
+ *
+ * try {
+ *     // do stuff
+ * } catch (ArithmeticException | Exception e) { // Violation, one of the exception
+ *                                                is declared to be illegal
+ *
+ * }
  *
+ * try {
+ *     // do stuff
+ * } catch (ArithmeticException e) { // OK
+ *
+ * }
+ * </pre>
+ * <p>
+ * To configure the check rejecting ArithmeticException catch also:
```

Comment:
`rejecting ArithmeticException catch also` -> `to extend default list with ArithmeticException`